### ISSUE TITLE
[ENH]: align_titles

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#22376]
### Problem

There's align_xlabels and align_ylabels to align the x and y labels, but no align_titles; this would be useful e.g. for
```python
from pylab import *
fig, axs = subplots(1, 2, subplot_kw={"xlabel": "x", "ylabel": "y", "title": "t"})
axs[0].imshow(zeros((3, 5)))
axs[1].imshow(zeros((5, 3)))
fig.align_labels()
```
![test](https://user-images.githubusercontent.com/1322974/152163429-d1169ce2-8818-4fb2-9a9b-545780a49a10.png)

### Proposed solution

Add Figure.align_titles.

**Comment by QuLogic** [comment]
This is _probably_ a good first issue; you have to copy what `align_xlabels` does, but using the title texts.

**Comment by pradhanvickey** [comment]
Is this still open? I'd like to try my hand at it.

**Comment by anntzer** [comment]
Go for it.

**Comment by pradhanvickey** [comment]
Thanks @anntzer . Will start working on it

**Comment by weiji-li** [comment]
@pradhanvickey are you still working on this issue?

**Comment by jklymak** [comment]
We tend not to reserve issues.  If there is no PR feel free to submit one.

**Comment by neil-gurnani** [comment]
@QuLogic / @anntzer. Both @lijake8 and myself would like to work on this. Is this okay?

Take

**Comment by lijake8** [comment]
take

**Comment by kkoutris** [comment]
Is this issue still open?
If yes can you assign me @anntzer ?

**Comment by anntzer** [comment]
See https://matplotlib.org/devdocs/devel/contributing.html#issues-for-new-contributors.

**Comment by kkoutris** [comment]
I don’t see any PR on the issue but probably @neil-gurnani  is still working on it. I’ll wait a couple of days or look for a different issue.

**Comment by aswarner** [comment]
I've been looking at this for a couple weeks -- it doesn't appear to be as easy as expected. We can replicate the grouping of axes with the the same rowspans, as done in align_xlabels(), however, updating the title location is done in the _AxesBase class, while the label updates are done in the Figure class. Importantly, the Figure class has access to subplots, but the axesBase class does not have a way to access the subplots or a grouper we make in the figure class, where align_titles() would exist. @anntzer  I'm new to Matplotlib, was wondering if there is some workaround I'm missing to share the relevant data between the classes.

Basically, we'd need to share either a grouper object or subplot location information from the Figure class (where the align_titles function would be written) to the axesBase class (where the title update occurs). Do you have any thoughts or are we missing anthing?

**Comment by jklymak** [comment]
the title moving logic was copied from the x/ylabel moving logic,  so aligning the titles should be possible after the automated moving that happens in `Axes`, just as its possible to align the x/y labels.

**Comment by jklymak** [comment]
I took a quick look at this again.  The logic in `axis.py` is exactly the same logic you would need to use in `_base.py` in `_update_title_position`.  You can just make another `_align_label_groups['titles']`.  I think there are some details to work out, but the information flow is exactly the same as for the way it is set up now.

**Comment by aswarner** [comment]
Thank you for the response! That's what I tried, but I don't see how the AxesBase class in` _base.py `has access to the `_align_label_groups['titles']` which is part of the FigureBase class.

**Comment by jklymak** [comment]
Each axes has a self.figure.

**Comment by aswarner** [comment]
@jklymak That helped! We figured it out and are working on testing and documentation

**Comment by christuangsit** [comment]
Hello, is this issue solved? If not, my friend and I would like to work on it.

**Comment by timhoffm** [comment]
The issue is open, so not solved yet. But as linked above, it has an associated open pull request, i.e. a solution proposal. Therefore it does not make sense to work on this issue. Please look for another topic if you want to contribute.

As a side question, we often see people who ask in the way you did above coming from a CS course? Is that true for you as well? If so, which course and did they give you instructions on how to find suitable issues, in particular identifying whether an issue has already an associated open pull request and to judge whether that pull request is still active?
